File: 01.calculator/seoyoung/calculator/src/test/java/CalculatorTest.java
import org.junit.jupiter.api.*;

import static org.assertj.core.api.Assertions.*;

class CalculatorTest {
    private Calculator calculator;

    @BeforeAll
    public void setUp() {
        calculator = new Calculator();
    }

    @Test
    void testStringToInteger() {
        assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(
                () -> {
                    calculator.stringToInteger("+");
                }
        ).withMessageMatching("숫자가 아닙니다.");
    }

    @Test
    void testStringToChar() {
        assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(
                () -> {
                    calculator.stringToChar("");
                }
        ).withMessageMatching("문자열이 비어있습니다.");
    }

    @Test
    void testAdd() {
        assertThat(calculator.add(4,3)).isEqualTo(7);
    }

    @Test
    void testSubtract() {
        assertThat(calculator.subtract(4,3)).isEqualTo(1);
    }

    @Test
    void testMultiply() {
        assertThat(calculator.multiply(4,3)).isEqualTo(12);
    }

    @Test
    void testDivide() {
        assertThat(calculator.divide(4,3)).isEqualTo(1);
    }

}

File: src/test/kotlin/calculator/CalculatorTest.kt
package calculator

import org.assertj.core.api.Assertions.assertThatExceptionOfType
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

class CalculatorTest {
    @Test
    fun testWhiteSpace() {
        assertThatExceptionOfType(IllegalArgumentException::class.java).isThrownBy {
            val calculator = Calculator(" ")
            calculator.expression
        }.withMessageMatching("Wrong input")
    }

    @Test
    fun testOperator() {
        assertThatExceptionOfType(IllegalArgumentException::class.java).isThrownBy {
            val calculator = Calculator("1 3 2")
            calculator.calculate()
        }.withMessageMatching("Not a valid operator")
    }

    @Test
    fun testAdd() {
        val expected = 5
        assertEquals(expected, Calculator.add(2, 3))
    }

    @Test
    fun testSubtract() {
        val expected = 1
        assertEquals(expected, Calculator.subtract(3, 2))
    }

    @Test
    fun testMultiply() {
        val expected = 6
        assertEquals(expected, Calculator.multiply(2, 3))
    }

    @Test
    fun testDivide() {
        val expected = 2
        assertEquals(expected, Calculator.divide(4, 2))
    }
}